Author: Nathan Phillip Brink <ohnobinki@ohnopublishing.net>
Purpose: Ensure that the CXXFLAGS environmental variable are properly communicated to cmake's unix driver.

--- a/CMakeLists.txt	2010-03-13 16:29:25.000000000 -0500
+++ b/CMakeLists.txt	2010-03-13 16:32:04.000000000 -0500
@@ -127,7 +127,7 @@
   add_definitions(${DEFAULT_COMPILE_WARNINGS_YES})
 endif(_comp_warnings STREQUAL "yes")
 if(_comp_warnings STREQUAL "kde")
-  set(CMAKE_CXX_FLAGS "${DEFAULT_COMPILE_WARNINGS_KDE} ${CMAKE_CXX_FLAGS}")
+  add_definitions(${DEFAULT_COMPILE_WARNINGS_KDE})
 endif(_comp_warnings STREQUAL "kde")
 
 
--- a/cmake/modules/PopplerMacros.cmake	Sat Mar 13 16:50:27 2010 -0500
+++ b/cmake/modules/PopplerMacros.cmake	Sat Mar 13 16:52:58 2010 -0500
@@ -91,7 +91,7 @@
   set(DEFAULT_COMPILE_WARNINGS_YES "-Wall -Wno-write-strings")
   set(DEFAULT_COMPILE_WARNINGS_KDE "-Wnon-virtual-dtor -Wno-long-long -Wundef -ansi -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -O2 -Wformat-security -Wmissing-format-attribute -fno-exceptions -fno-check-new -fno-common")
 
-  set(CMAKE_CXX_FLAGS                "-Woverloaded-virtual")
+  add_definitions(-Woverloaded-virtual)
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
   set(CMAKE_CXX_FLAGS_RELEASE        "-O2 -DNDEBUG")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
